### PR TITLE
add component pages, move files and add 2 more chapters

### PR DESCRIPTION
**Pull request by DessertSource**
Add 2 more chapters named Essential Features, Components and their own pages. Also moved images to a centralized images folder. Hopefully this file moving will be the last one I have to do on the repository.

### CODE REVIEW

**Comment by Sixmorphugus** [comment]
Can you add temp brick selection (I think it's middle click? Check) to the catalog page, as well as click and drag for both the item and brick catalog?

**Comment by qoh** [comment]
I'd really prefer _not_ to have capitalized folder and file names (like `images/Tools`). It goes against URL and most folder structure naming conventions.

Could you also squash some of the redundant commits? `ffc9a63`, `c9745f7`, and `0a1f589` for example shouldn't be separate.

**Review by qoh — CHANGES_REQUESTED** [review]
I'd really prefer _not_ to have capitalized folder and file names (like `images/Tools`). It goes against URL and most folder structure naming conventions (which tend to be like `path/to/my-file.png` or `path/to/my_file.png`).

Could you also squash some of the redundant commits? `ffc9a63`, `c9745f7`, and `0a1f589` for example shouldn't be separate.

File: src/SUMMARY.md [hunk 1]
```diff
@@ -1,8 +1,9 @@
 # Summary
 
 - [Brickadia](./brickadia.md)
+- [System Requirements]()
```
Comment (qoh) [inline src/SUMMARY.md:4]: Did you intend to leave this link empty? It's fine, but I thought I saw you had some system requirements written already. There's also some on the website and the Discord store. Either way, I've started tracking the need for the page in #11.

File: src/components/audio.md [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+
+![Icon](../images/Components/Audio.png)
+
+This component can add sounds or music to bricks.
+
+![Edit Menu](../images/Components/EditMenuAudio.png)
+
+## Basic Settings
+
+* **Pickup** - This determines the sound that plays from the brick.
```
Comment (qoh) [inline src/components/audio.md]: Copy/paste error?

File: src/components/item_spawn.md [hunk 3]
```diff
@@ -0,0 +1,33 @@
...
+* **Auto Disable** - If this is enabled, this will disable the pick-up when someone picks up the item.
+* **Respawn Time** - This determines the waiting time for an item to be able to be picked up.
+* **Respawns on Round Start** - This determines if the item can be picked up instantly when a new round starts.
+* **Delay After Round Start** - This determines the delay period before you can pick up the item.
+
+## Positioning
+* **Offset Direction** - Direction that the item will be shown, relative to the brick.
+* **Offset Distance** - Distance between the item itself and the brick with this component.
+* **Rotation** - The rotation of the item.
+* **Scale** - The scale of the spawned item.
```
Comment (qoh) [inline src/components/item_spawn.md:22]: The explanations of these seem pretty unhelpful.

File: src/components/item_spawn.md [hunk 4]
```diff
@@ -0,0 +1,33 @@
...
+## Animation
+
+Animation is a toggleable property.
+
+* **Animation Axis** - The axis that the item will spin on.
+* **Use Relative Axis** - The relative axis that the item will spin on.
+* **Spin Speed** - The speed at which the item spins.
+* **Bob Speed** - The speed at which the item bobs vertically.
+* **Bob Height** - The maximum height of the item's vertical bobbing.
+* **Phase** - The phase of the animation.
```
Comment (qoh) [inline src/components/item_spawn.md:33]: The floor here is made of floor

**Comment by qoh** [comment]
Many of the explanations for component properties suffer from 'the floor here is made of floor'-like problems, such as:
* Light / Brightness - The brightness of the light
* Animation / Phase -  The phase of the animation.
* Item Spawn / Rotation - The rotation of the item.
* Item Spawn / Scale - The scale of the spawned item.


These can always be improved later though, so don't worry about them.

File: src/components/audio.md (hunk 2 above)
Comment (DessertSource) [reply src/components/audio.md]: Yes, this was a copy/paste error that I had overlooked.

File: src/SUMMARY.md (hunk 1 above)
Comment (DessertSource) [reply src/SUMMARY.md:4]: Yes, I did. I will add the link when the next pull request comes around.

**Review by DessertSource — COMMENTED** [review]
I'll get around to fixing the typo. For the "the floor here is made of floor" problems, I intend to fix them by adding additional example images or rephrasing the descriptions in the next pull request.

**Review by qoh — DISMISSED** [review]

File: src/chapter_2.md [hunk 5]
```diff
@@ -0,0 +1,7 @@
+# Chapter 2: Essential Features
+Brickadia has an array of "quality of life" features, if those didn't exist, you wouldn't be able to do things easily.
```
Comment (Sixmorphugus) [inline src/chapter_2.md]: Slight run-on scentence here, consider replacing first ","

**Review by Sixmorphugus — COMMENTED** [review]
Excellent stuff!

File: src/chapter_2.md (hunk 5 above)
Comment (qoh) [reply src/chapter_2.md]: What about something similar this?
> Brickadia has many quality of life features designed to make a lot of common things easier.

I agree that the way it's written now is very odd.

File: src/essentials/radial_menus.md [hunk 6]
```diff
@@ -0,0 +1,24 @@
...
+
+![Radial Emote Menu](../images/essentials/radial_emote_menu.png)
+
+To use an emote, hold the **H** key to open the menu and hover over the emote you want. From there, release the **H** key.
+
+To stop using an emote (hugging and sitting), hover over the X icon.
+
+## Radial Tool Menu
+
+![Radial Tool Menu](../images/essentials/radial_tool_menu.png)
```
Comment (Zeblote) [inline src/essentials/radial_menus.md:15]: retake screenshot with higher graphics quality

File: src/chapter_2.md (hunk 5 above)
Comment (Zeblote) [reply src/chapter_2.md]: sentence must be rewritten like the above discussions suggested (I thought this comment would show up in the chain but apparently that's not how review works)

File: src/chapter_2.md [hunk 7]
```diff
@@ -0,0 +1,7 @@
...
+
+1. [Catalog](essentials/catalog.md) - A user interface where you get your bricks or weapons from.
+2. [Radial Menus](essentials/radial_menus.md) - A user interface that you can get your tools or use emotes from.
+3. [Presets](essentials/presets.md) - A feature that's almost everywhere. Very useful for palettes, graphics settings, and more.
+4. [Quickbar](essentials/quickbar.md) - Your main hotbar in Brickadia.
```
Comment (Zeblote) [inline src/chapter_2.md]: it is called quickbar, not hotbar

File: src/essentials/quickbar.md [hunk 8]
```diff
@@ -0,0 +1,33 @@
+# Quickbar
+
+The Quickbar is the hotbar used for this game.
```
Comment (Zeblote) [inline src/essentials/quickbar.md]: it is called quickbar, not hotbar

File: src/essentials/catalog.md [hunk 9]
```diff
@@ -0,0 +1,21 @@
+# Catalog
+
+The catalog is where you can grab your bricks and weapons from. It is a user interface for getting items.
```
Comment (Zeblote) [inline src/essentials/catalog.md]: writing "it is a user interface" in the second sentence is very strange

File: src/essentials/presets.md [hunk 10]
```diff
@@ -0,0 +1,28 @@
+# Presets
+
+In Brickadia, presets are everywhere.
+
+Presets can be saved as:
```
Comment (Zeblote) [inline src/essentials/presets.md]: saved for

File: src/essentials/presets.md [hunk 11]
```diff
@@ -0,0 +1,28 @@
...
+- Avatars
+- Palettes
+- Environment configurations
+- Graphics settings
```
Comment (Zeblote) [inline src/essentials/presets.md]: all settings

File: src/essentials/presets.md (hunk 10 above)
Comment (Zeblote) [reply src/essentials/presets.md]: missing components

Comment (Zeblote) [reply src/essentials/presets.md]: missing minigames

File: src/essentials/presets.md [hunk 12]
```diff
@@ -0,0 +1,28 @@
...
+![Preset Menu](../images/essentials/presets_menu.png)
+
+When you hover over a preset in this menu, 3 options will show.
+
+![Preset Menu](../images/essentials/presets_menu_hovered_over.png)
+
+Here's a list of the options (in order):
+1. Rename - Click this button to rename the preset.
+2. Overwrite - Click this yellow button to overwrite the preset with your current configuration.
+3. Delete - Click this red button to remove the preset. A prompt that lets you confirm the deletion will appear.
```
Comment (Zeblote) [inline src/essentials/presets.md]: prompt also appears for overwrite, yet is only mentioned here

Comment (Zeblote) [reply src/essentials/presets.md]: just remove the mention, people will see the prompt for themselves

File: src/essentials/quickbar.md [hunk 13]
```diff
@@ -0,0 +1,33 @@
...
+![Item Quickbar](../images/essentials/item_quickbar.png)
+
+This can be configured through [Minigames]().
+
+## Default Keybindings
+
+|Action|Keybinding|Functionality|
+|---|---|---|
+|Slot 1-9*|1 - 9|Equips the item in slot 1 - 9.|
+|Slot 10|0|Equips the item in slot 10.|
```
Comment (Zeblote) [inline src/essentials/quickbar.md]: writing `Slot 1-10 | 1 - 0` is acceptable

File: src/components/spot_light.md [hunk 14]
```diff
@@ -0,0 +1,24 @@
+# Spot Light
+
+![Icon](../images/components/spot_light.png)
+
+This component adds a spotlight to a brick.
```
Comment (Zeblote) [inline src/components/spot_light.md:7]: needs an image of what that looks like in world

File: src/components/point_light.md [hunk 15]
```diff
@@ -0,0 +1,22 @@
+# Point Light
+
+![Icon](../images/components/point_light.png)
+
+This component adds a point light to a brick.
```
Comment (Zeblote) [inline src/components/point_light.md:7]: needs an image of what that looks like in world

File: src/components/item_spawn.md [hunk 16]
```diff
@@ -0,0 +1,33 @@
+# Item Spawn
+
+![Icon](../images/components/item_spawn.png)
+
+This component can spawn weapons and adjust their pick-up appearance. Compatible with [Minigames]().
```
Comment (Zeblote) [inline src/components/item_spawn.md:7]: needs an image of what that looks like in world

File: src/SUMMARY.md [hunk 17]
```diff
@@ -12,11 +13,22 @@
   - [Applicator](tools/applicator.md)
   - [Outliner](tools/outliner.md)
 
+- [Essential Features](chapter_2.md)
+  - [Catalog](essentials/catalog.md)
+  - [Radial Menus](essentials/radial_menus.md)
+  - [Presets](essentials/presets.md)
+  - [Quickbar](essentials/quickbar.md)
+
+- [Components](chapter_3.md)
```
Comment (Zeblote) [inline src/SUMMARY.md:22]: for all of these, it seems useless to copy their settings and tooltips to the book - someone will have to constantly update this as we update the game and it gives the user no new information

Comment (Zeblote) [reply src/SUMMARY.md:22]: instead we could be giving users tips on how to use the components more effectively or something like that

**Review by Zeblote — CHANGES_REQUESTED** [review]

**Review by qoh — CHANGES_REQUESTED** [review]
`catalog_bricks.png` seems to be swapped with `catalog_items.png`

**Review by qoh — DISMISSED** [review]

**Comment by Sixmorphugus** [comment]
This PR stuff is tricky to figure out how to use
I went and resolved all the conversations around things that were fixed (that run-on scentence and using Quickbar instead of hotbar)

Requested zeb look at the changes to make sure he's happy.

**Review by Sixmorphugus — DISMISSED** [review]

File: src/components/point_light.md [hunk 18]
```diff
@@ -8,17 +8,11 @@ This component adds a point light to a brick.
...
-
-## Point Light
-
-* **Match Brick Shape** - Matches the point light shape with the brick cubically.
\ No newline at end of file
+## Tips for this Component
+
+- You can use small radius point lights to light up signs, glowing crystals, neon lights, etc.
+- Point lights without shadows can cause undesired light leaks. It is advisable to turn down the *light radius* when this happens.
+![Light Leak](../images/components/light_leak.png)
```
Comment (Zeblote) [inline src/components/point_light.md:16]: bad formatting
image

File: src/components/point_light.md [hunk 19]
```diff
@@ -8,17 +8,11 @@ This component adds a point light to a brick.
...
+- Do not overlap point lights too much as this can cause performance issues.
+- It is important to use the *Cast Shadows* option sparsely or when it's necessary. Misuse of this can lead to your point lights casting abnormal shadows like this.
+![Shadow Example](../images/components/point_light_shadow_example_1.png)
```
Comment (Zeblote) [inline src/components/point_light.md:22]: needs mention of extreme performance cost of point light shadows

**Review by Zeblote — CHANGES_REQUESTED** [review]
almost ready now

**Review by Zeblote — APPROVED** [review]

**Review by Sixmorphugus — APPROVED** [review]

**Pull request merged**